src/NHibernate/Impl/SessionFactoryImpl.cs

```diff
@@ -580,6 +586,7 @@ public IStatelessSession OpenStatelessSession(DbConnection connection)
 
 		public IEntityPersister GetEntityPersister(string entityName)
 		{
+			CheckNotDisposed();
 			IEntityPersister value;
 			if (entityPersisters.TryGetValue(entityName, out value) == false)
```

Comment:
`entityPersisters` have their cache property destroyed by the factory closing, so guarding against their use. This also causes most session usages to throw if their session factory is disposed, which is also the aim of this PR.